include/swift/AST/SemanticAttrs.def

```diff
@@ -127,8 +127,21 @@ SEMANTICS_ATTR(LIFETIMEMANAGEMENT_COPY, "lifetimemanagement.copy")
 
 SEMANTICS_ATTR(NO_PERFORMANCE_ANALYSIS, "no_performance_analysis")
 
-// A flag used to turn off moveonly diagnostics on functions that allocbox to
-// stack specialized.
+// A flag used to turn off moveonly diagnostics on a function due to an earlier
+// pass that ran.
+//
+// DISCUSSION: This is used in a few situations:
+//
+// 1. When allocbox to stack specializes a function, we do not want to emit move
+// errors twice, once on the specialized and once on the original
+// function. Instead, we put this on the original function.
+//
+// 2. If we emit a diagnose invalid escaping captures error due to an inout
+// being escaped into an escaping closure, we do not want to emit move errors in
+// the closure. This is because SILGen today assumes that we will error in such
+// cases and thus does not emit markers in said function for the inout. This
+// then causes us to emit spurious "found a copy of a noncopyable value" errors
+// that may cause the user to think there is a bug in the compiler.
 SEMANTICS_ATTR(NO_MOVEONLY_DIAGNOSTICS, "sil.optimizer.moveonly.diagnostic.ignore")
```

Comment:
Can you add a regression test for this?

Reply:
I'm going to add the regression test in a follow on commit. I am adding in a follow on commit a large suite of closure tests. The reason why I am not including it here is that it crashes on main due to a different issue that I am going to fix.